active/teachingtip/TeachingTipSpec.md

```diff
@@ -0,0 +1,565 @@
+
+XAML
+```XAML
+<Button x:Name="SaveButton" Content="Save">
+    <Button.Resources>
+        <muxc:TeachingTip x:Name="AutoSaveTip"
+            Target="{x:Bind SaveButton}"
+            Title="Save automatically"
+            Subtitle="When you save your file to OneDrive, we save your changes as you go - so you never have to.">
+        </muxc:TeachingTip>
+    </Button.Resources>
+</Button>
+```
+
+C#
+```C#
+public MainPage()
+{
+    this.InitializeComponent();
+
+    if(!HaveExplainedAutoSave())
+    {
+        AutoSaveTip.IsOpen = true;
+        SetHaveExplainedAutoSave();
+    }
+}
+```
+
+Here's the result when this Page containing the button and teaching tip shown:
+
+![A sample app with a teaching tip pointing at the save button. The tip title reads "Saving automatically" and the subtitle reads "We save your changes as you go - so you never have to." There is a close button on the top right corner of the teaching tip.](TeachingTipPointerSample.jpg)
+
+### Non-pointing tips
+
+Not all tips relate to an element onscreen. For these scenarios, do not set the Target property and the teaching tip will instead display relative to the edges of the XamlRoot. However, a teaching tip can have the pointer removed while retaining placement relative to a UI element by setting the PointerMode property to "Off". 
+
+Unlike pointing teaching tips which can only be added to a resource dictionary, non-pointing teaching tips can also be added through the element tree, as shown below.
+
+XAML
+```XAML
+<Button x:Name="SaveButton" Content="Save" />
+
+<muxc:TeachingTip x:Name="AutoSaveTip"
+    Title="Saving automatically"
+    Subtitle="We save your changes as you go - so you never have to.">
+</muxc:TeachingTip>
+```
+
+Note that in this example the TeachingTip is in the element tree rather than in a ResourceDictionary. This has no effect on behavior; the TeachingTip only displays when opened, and takes up no layout space.
+![A sample app with a teaching tip in the bottom right corner. The tip title reads "Saving automatically" and the subtitle reads "We save your changes as you go - so you never have to." There is a close button on the top right corner of the teaching tip.](TeachingTipSampleApp.jpg)
+
+### Add an offset 
+
+You can control how far a pointing teaching tip is set apart from its target and how far a non-pointing teaching tip is set apart from the the edges of the XamlRoot using the TargetOffset property. Like [Margin](https://docs.microsoft.com/en-us/dotnet/api/system.windows.frameworkelement.margin?view=netframework-4.7.2), TargetOffset has four values – left, right, top, and bottom – so only the relevant values are used. For example, if the target is left of the teaching tip, the TargetOffset.Left will be used. If the teaching tip has no target, the TargetOffset’s Right and Bottom will be used to space it from the bottom right corner of the XamlRoot.  
```

Comment:
The name "TargetOffset" is strange. It's not the offset of the target. And it's not the offset of the tip relative to the target. And it's not the offset of the tip relative to its default placement. Here, the TargetOffset is positive 80, so I expect that to mean that the tip is 80 pixels southeast of "something", but instead it moves northwest from the corner of the XAML root. Should we just call this `Margin`? Because it describes "how much space to put between the tip and its reference point."

Reply:
Resolved [here](https://github.com/Microsoft/microsoft-ui-xaml-specs/commit/5be7d4bab21c5c74909927055faaa81fa2eff706).